lib/bsdlib/nrf91_sockets.c

```diff
@@ -737,6 +737,9 @@ static ssize_t nrf91_socket_offload_recvfrom(int sd, void *buf, short int len,
 static ssize_t nrf91_socket_offload_send(int sd, const void *buf, size_t len,
 					 int flags)
 {
+	if (IS_ENABLED(CONFIG_NRF91_SOCKET_SEND_SPLIT_LARGE_BLOCKS)) {
+		len = MIN(len, CONFIG_NRF91_SOCKET_BLOCK_LIMIT);
```

Comment:
Theoretically it should only be allowed for stream sockets, it doesn't make sense to split a datagram. But I've rarely seen datagrams as large as 2k, so I guess it's fine for a workaround. It would be tricky to track socket types at this layer. it should be handled by bsdlib. 

Is there a Jira issue open for this bug?

Reply:
Yes, I mentioned it in the opening comment: "Jira: NRF91-601"  :-)

Reply:
I agree, should be handled in bsdlib or in the modem.  This was just to prove that such an approach solves a real problem a customer has.